### DISCUSSION TITLE
How can I use Isar from a remote saver?

### DISCUSSION THREAD

**Discussion by Tembocs** [isar/isar#158] [Help] [answered]
Is there a way Isar can be used from a remote saver? For example, having data saved in a remove saver with Isar installed that can be accessed by mobile app from devices. Specifically, suppose I need to create a students database, and that students with an app that can access the database are able to connect to it automatically and see their respective information from the database, is there a way I can do this with Isar?

**Comment by MarcelGarus** [comment, marked as answer]
You should probably design a designated API around the database that the devices will use.

Having the database directly accessible to devices is usually not considered best practice, because devices then depend on the internal structure of the data storage, which means you can't change how data is stored without also changing how it is accessed from the outside. Basically, this makes versioning APIs very difficult.

Also, note that if you also want users to be able to change data in the database makes things even complicated. Something akin to Firestore security rules may be used to restrict the permitted changes, but having well-tested, actual _code_ make this decision is easier most of the time.

**Reply by Tembocs** [reply]
I totally agree. I actually mean the a database that can be used in the liking of Firebase but much more easier to the end users (developers in this case).
